pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -1147,21 +1152,62 @@ func (r *Builder) getVolumeAndAccessMode(storageClassName string) ([]core.Persis
 
 }
 
-// Get the OpenstackVolumePopulator CustomResource based on the image name.
-func (r *Builder) getVolumePopulatorCR(name string) (populatorCr api.OpenstackVolumePopulator, err error) {
-	populatorCr = api.OpenstackVolumePopulator{}
-	err = r.Destination.Client.Get(context.TODO(), client.ObjectKey{Namespace: r.Plan.Spec.TargetNamespace, Name: name}, &populatorCr)
+// Get the OpenstackVolumePopulator CustomResource based on the image ID.
+func (r *Builder) getVolumePopulatorCR(imageID string) (populatorCr api.OpenstackVolumePopulator, err error) {
+	populatorCrList := &api.OpenstackVolumePopulatorList{}
+	err = r.Destination.Client.List(context.TODO(), populatorCrList, &client.ListOptions{
+		Namespace: r.Plan.Spec.TargetNamespace,
+		LabelSelector: labels.SelectorFromSet(map[string]string{
+			"migration": getMigrationID(r.Context),
+			"imageID":   imageID,
+		}),
+	})
+	if err != nil {
+		return
+	}
+	if len(populatorCrList.Items) == 0 {
+		err = k8serr.NewNotFound(api.SchemeGroupVersion.WithResource("OpenstackVolumePopulator").GroupResource(), imageID)
+		return
+	}
+	if len(populatorCrList.Items) > 1 {
+		err = liberr.New("multiple OpenstackVolumePopulator CRs found for image", "imageID", imageID)
```

Comment:
did you mean to proceed with setting populatorCr below? if we expect the caller to fail, maybe we should keep populatorCr nil?